File: 4.cpp
#include<iostream>
#include<algorithm>

using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int mid = (nums1.size() + nums2.size()) / 2;
        int i = 0, j = 0;
        vector<int> nums;
        while(nums.size() <= mid) {
            if (i < nums1.size() && j < nums2.size())
                nums.push_back(nums1[i] < nums2[j] ? nums1[i++] : nums2[j++]);
            else if (i < nums1.size()) 
                nums.push_back(nums1[i++]);
            else nums.push_back(nums2[j++]);
        }
        return (nums1.size() + nums2.size()) % 2 == 0 ? (nums[mid-1] + nums[mid]) / 2.0 : nums[mid];
    }
};

int main() {
    Solution s;
    int n1, n2;
    cin >> n1 >> n2;
    vector<int> nums1(n1), nums2(n2);
    for (int i = 0; i < n1; i++) {
        cin >> nums1[i];
    }
    for (int i = 0; i < n2; i++) {
        cin >> nums2[i];
    }
    cout << s.findMedianSortedArrays(nums1, nums2) << endl;
    return 0;
}

File: 4.py
class Solution:
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        mid = (len(nums1) + len(nums2)) // 2
        merged = []
        i = 0
        j = 0
        while len(merged) <= mid:
            if i < len(nums1) and j < len(nums2):
                if nums1[i] <= nums2[j]:
                    merged.append(nums1[i])
                    i += 1
                else:
                    merged.append(nums2[j])
                    j += 1
            elif i < len(nums1):
                merged.append(nums1[i])
                i += 1
            else:
                merged.append(nums2[j])
                j += 1
        return (merged[-1] + merged[-2]) / 2 if (len(nums1) + len(nums2)) % 2 == 0 else merged[-1]
        
        
if __name__ == '__main__':
    s = Solution()
    nums1 = [int(x) for x in input().split()]
    nums2 = [int(x) for x in input().split()]
    print(s.findMedianSortedArrays(nums1, nums2))